### PR TITLE
 Thêm chức năng đăng ký ghế ngồi khi đăng ký lịch làm việc

### PR DESCRIPTION
**Pull request by phanvantan73**

### CODE REVIEW

File: app/Models/Seat.php [hunk 1]
```diff
@@ -14,6 +14,30 @@ class Seat extends Model
...
+        return $this->belongsToMany(User::class)
+            ->withPivot('id', 'registration_date', 'is_full_day');
+    }
+
+    public function checkUsedSeat($date, $shift)
+    {
+        switch ($shift) {
+            case config('site.shift.all'):
+                return $this->users()->wherePivot('registration_date', $date)
+                    ->wherePivotIn('shift', [$shift, config('site.shift.morning'), config('site.shift.afternoon'),])
```
Comment (framgiaciv3) [inline app/Models/Seat.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Seat.php</br>Comma not allowed after last value in single-line array declaration

File: app/Models/Seat.php [hunk 2]
```diff
@@ -14,6 +14,30 @@ class Seat extends Model
...
+                    ->exists();
+            case config('site.shift.morning'):
+                return $this->users()->wherePivot('registration_date', $date)
+                    ->wherePivotIn('shift', [config('site.shift.all'), $shift,])
```
Comment (framgiaciv3) [inline app/Models/Seat.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Seat.php</br>Comma not allowed after last value in single-line array declaration

File: app/Models/Seat.php [hunk 3]
```diff
@@ -14,6 +14,30 @@ class Seat extends Model
...
+                    ->exists();
+            case config('site.shift.afternoon'):
+                return $this->users()->wherePivot('registration_date', $date)
+                    ->wherePivotIn('shift', [config('site.shift.all'), $shift,])
```
Comment (framgiaciv3) [inline app/Models/Seat.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.CommaAfterLast</br>**Severity**: error</br>**File**: app/Models/Seat.php</br>Comma not allowed after last value in single-line array declaration

File: app/Http/Controllers/WorkScheduleController.php [hunk 4]
```diff
@@ -40,27 +49,70 @@ public function index()
...
-                ],
-                [
-                    'location_id' => $data['location'][$key],
-                    'shift' => $value,
-                ]
-            );
+        try {
+            $data = $request->all();
+            foreach ($data['shift'] as $key => $value) {
+                DB::transaction(function () use ($request, $data, $key, $value) {
```
Comment (framgiaciv3) [inline app/Http/Controllers/WorkScheduleController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.ContentAfterOpenBracket</br>**Severity**: error</br>**File**: app/Http/Controllers/WorkScheduleController.php</br>Opening parenthesis of a multi-line function call must be the last content on the line

File: app/Http/Controllers/WorkScheduleController.php [hunk 5]
```diff
@@ -40,27 +49,70 @@ public function index()
...
+                    $request->user()->workSchedules()->updateOrCreate(
```
Comment (framgiaciv3) [inline app/Http/Controllers/WorkScheduleController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.Indent</br>**Severity**: error</br>**File**: app/Http/Controllers/WorkScheduleController.php</br>Multi-line function call not indented correctly; expected 8 spaces but found 12

Comment (framgiaciv3) [inline app/Http/Controllers/WorkScheduleController.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.Methods.FunctionCallSignature.CloseBracketLine</br>**Severity**: error</br>**File**: app/Http/Controllers/WorkScheduleController.php</br>Closing parenthesis of a multi-line function call must be on a line by itself

File: app/Http/Controllers/WorkScheduleController.php [hunk 6]
```diff
@@ -40,27 +49,77 @@ public function index()
...
+        $seats = Auth::user()->workspace->seats()->pluck('seats.name', 'seats.id');
         $data = $this->workingSchedule->getUserSchedule(Auth::user()->id);
         $dataLocation = $this->workingSchedule->getLocation(Auth::user()->id);
+        $dataSeat = Auth::user()->seats()->pluck('seat_id', 'registration_date');
+
+        if (count($data) == config('site.disallow_register')) {
+            $registered = false;
+        } else {
+            $registered = true;
+        }
```
Comment (tranducquoc) [inline app/Http/Controllers/WorkScheduleController.php]: dùng toán tử 3 ngôi cho xử lý logic này

File: app/Models/Seat.php [hunk 7]
```diff
@@ -14,6 +14,30 @@ class Seat extends Model
...
+                    ->exists();
+            case config('site.shift.morning'):
+                return $this->users()->wherePivot('registration_date', $date)
+                    ->wherePivotIn('shift', [config('site.shift.all'), $shift])
+                    ->exists();
+            case config('site.shift.afternoon'):
+                return $this->users()->wherePivot('registration_date', $date)
+                    ->wherePivotIn('shift', [config('site.shift.all'), $shift])
+                    ->exists();
+        }
```
Comment (tranducquoc) [inline app/Models/Seat.php:43]: không có default à em?

File: app/Http/Controllers/WorkScheduleController.php (hunk 6 above)
Comment (phanvantan73) [reply app/Http/Controllers/WorkScheduleController.php]: ready

File: app/Models/Seat.php (hunk 7 above)
Comment (phanvantan73) [reply app/Models/Seat.php:43]: ready

**Pull request closed without merging**